### ISSUE TITLE
Outputs of torch.ceil abnormally mismatch on GPU when adding an intermediate result as output

### ISSUE THREAD

**Issue by Azyka** [NVIDIA/TensorRT#3472]
## Description

When outputting an intermediate result of `torch.sigmoid` in this model:
<img width="135" alt="image" src="https://github.com/NVIDIA/TensorRT/assets/74590664/d6bd9003-fae2-4b9b-8453-8e766398ac90">

New:
<img width="176" alt="image" src="https://github.com/NVIDIA/TensorRT/assets/74590664/9ea1caa8-2abb-4da9-b8b1-c6f758788234">

The output of `torch.ceil` is expected to be the same for the same input in this 2 graphs. However, it mismatched bewteew the 2 models.

## Environment

**TensorRT Version**: 9.1.0.post12.dev4

**NVIDIA GPU**: RTX 1660

**NVIDIA Driver Version**: 525.147.05

**CUDA Version**: 12.0

**CUDNN Version**: 8.9.4.25

Operating System: Ubuntu 22.04.3 LTS (x86_64)

Python Version (if applicable): 3.10.12

Tensorflow Version (if applicable): 2.13.0

PyTorch Version (if applicable): 2.1.0+cu118

## Relevant Files

**Model link**: 
[models.zip](https://github.com/NVIDIA/TensorRT/files/13413058/models.zip)

## Steps To Reproduce

**Script**:
```python
import numpy as np
from numpy import testing
import torch
from polygraphy.backend.trt import EngineFromNetwork, NetworkFromOnnxPath, TrtRunner

DEVICE='cuda'

class Model0(torch.nn.Module):
    def __init__(self):
        super().__init__()

    def forward(self, *args):
        _args = args
        getitem = _args[0];  _args = None
        tan = torch.tan(getitem);  getitem = None
        sigmoid = torch.sigmoid(tan);  tan = None
        ceil = torch.ceil(sigmoid);  sigmoid = None
        return (ceil,)

model_0 = Model0()
output_names_0 = ['v5_0']
data_0 = np.random.normal(5, 1, size=(35,6,3)).astype(np.float16)
input_dict_0 = {'v0_0': data_0}
inputs_0 = tuple(torch.from_numpy(v).to(DEVICE) for _, v in input_dict_0.items())
torch.onnx.export(model_0, inputs_0, '0.onnx', verbose=False, input_names=['v0_0'], output_names=output_names_0, opset_version=14, do_constant_folding=False)

class Model1(torch.nn.Module):
    def __init__(self):
        super().__init__()

    def forward(self, *args):
        _args = args
        getitem = _args[0]
        tan = torch.tan(getitem)
        sigmoid = torch.sigmoid(tan)
        ceil = torch.ceil(sigmoid)
        return (sigmoid, ceil)

model_1 = Model1()
output_names_1 = ['v1_0', 'v5_0']
input_dict_1 = input_dict_0
inputs_1 = tuple(torch.from_numpy(v).to(DEVICE) for _, v in input_dict_1.items())
torch.onnx.export(model_1, inputs_1, '1.onnx', verbose=False, input_names=['v0_0'], output_names=output_names_1, opset_version=14, do_constant_folding=False)

engine_0 = EngineFromNetwork(NetworkFromOnnxPath('0.onnx'))
with TrtRunner(engine_0) as runner:
    output_0 = runner.infer(input_dict_0)

engine_1 = EngineFromNetwork(NetworkFromOnnxPath('1.onnx'))
with TrtRunner(engine_1) as runner:
    output_1 = runner.infer(input_dict_1)
output_name_dict = {'v5_0': 'v5_0'}

print('=========================')
try:
    for tensor_name_0, tensor_name_1 in output_name_dict.items():
        testing.assert_allclose(output_0[tensor_name_0], output_1[tensor_name_1], rtol=1, err_msg=f'at {tensor_name_0}, {tensor_name_1}')
    print("tensorRT does not trigger assertion")
except AssertionError as e:
    print("tensorRT triggers assertion")
    print(e)
print('=========================')

```

Steps to repro:
1. Run the script

Output assertion:
```
=========================
tensorRT triggers assertion

Not equal to tolerance rtol=1, atol=0
at v5_0, v5_0
Mismatched elements: 16 / 630 (2.54%)
Max absolute difference: 1.
Max relative difference: 0.
 x: array([[[1., 1., 1.],
        [1., 1., 1.],
        [1., 1., 1.],...
 y: array([[[1., 1., 1.],
        [1., 1., 1.],
        [1., 1., 1.],...
=========================
```

**Have you tried [the latest release](https://developer.nvidia.com/tensorrt)?**: Yes

**Can this model run on other frameworks?** For example run ONNX model with ONNXRuntime (`polygraphy run <model.onnx> --onnxrt`): Yes

**Comment by zerollzeng** [comment]
Looks like it's fixed in internal release, please wait for the 9.2 comes out.
```
[I] Accuracy Comparison | trt-runner-N0-11/24/23-09:50:10 vs. onnxrt-runner-N0-11/24/23-09:50:10
[I]     Comparing Output: 'v1_0' (dtype=float16, shape=(35, 6, 3)) with 'v1_0' (dtype=float16, shape=(35, 6, 3))
[I]         Tolerance: [abs=1e-05, rel=1e-05] | Checking elemwise error
[I]         trt-runner-N0-11/24/23-09:50:10: v1_0 | Stats: mean=0.64425, std-dev=0.092853, var=0.0086217, median=0.64038, min=0.5 at (0, 0, 2), max=0.82471 at (6, 0, 1), avg-magnitude=0.64425
[I]         onnxrt-runner-N0-11/24/23-09:50:10: v1_0 | Stats: mean=0.64425, std-dev=0.092853, var=0.0086217, median=0.64038, min=0.5 at (0, 0, 2), max=0.82471 at (6, 0, 1), avg-magnitude=0.64425
[I]         Error Metrics: v1_0
[I]             Minimum Required Tolerance: elemwise error | [abs=0] OR [rel=0] (requirements may be lower if both abs/rel tolerances are set)
[I]             Absolute Difference | Stats: mean=0, std-dev=0, var=0, median=0, min=0 at (0, 0, 0), max=0 at (0, 0, 0), avg-magnitude=0
[I]             Relative Difference | Stats: mean=0, std-dev=0, var=0, median=0, min=0 at (0, 0, 0), max=0 at (0, 0, 0), avg-magnitude=0
[I]         PASSED | Output: 'v1_0' | Difference is within tolerance (rel=1e-05, abs=1e-05)
[I]     Comparing Output: 'v5_0' (dtype=float16, shape=(35, 6, 3)) with 'v5_0' (dtype=float16, shape=(35, 6, 3))
[I]         Tolerance: [abs=1e-05, rel=1e-05] | Checking elemwise error
[I]         trt-runner-N0-11/24/23-09:50:10: v5_0 | Stats: mean=1, std-dev=0, var=0, median=1, min=1 at (0, 0, 0), max=1 at (0, 0, 0), avg-magnitude=1
[I]         onnxrt-runner-N0-11/24/23-09:50:10: v5_0 | Stats: mean=1, std-dev=0, var=0, median=1, min=1 at (0, 0, 0), max=1 at (0, 0, 0), avg-magnitude=1
[I]         Error Metrics: v5_0
[I]             Minimum Required Tolerance: elemwise error | [abs=0] OR [rel=0] (requirements may be lower if both abs/rel tolerances are set)
[I]             Absolute Difference | Stats: mean=0, std-dev=0, var=0, median=0, min=0 at (0, 0, 0), max=0 at (0, 0, 0), avg-magnitude=0
[I]             Relative Difference | Stats: mean=0, std-dev=0, var=0, median=0, min=0 at (0, 0, 0), max=0 at (0, 0, 0), avg-magnitude=0
[I]         PASSED | Output: 'v5_0' | Difference is within tolerance (rel=1e-05, abs=1e-05)
[I]     PASSED | All outputs matched | Outputs: ['v1_0', 'v5_0']
[I] Accuracy Summary | trt-runner-N0-11/24/23-09:50:10 vs. onnxrt-runner-N0-11/24/23-09:50:10 | Passed: 1/1 iterations | Pass Rate: 100.0%
[I] PASSED | Runtime: 19.464s | Command: /home/scratch.zeroz_sw/miniconda3/bin/polygraphy run 1.onnx --trt --onnxrt

```

**Comment by ttyio** [comment]
The 9.2 is released in https://github.com/NVIDIA/TensorRT/tree/release/9.2, I will close this since the last comment show it is fixed in 9.2, thanks all!
